File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/facade/FacadePattern.java
package jv.com.tencent.cain.design_pattern.facade;

/**
 * @author : jiangyu
 * @date : 2020/2/5
 * @desc : 外观设计模式
 */
public class FacadePattern {
    public static void main(String[] args) {
        FacadeSystem facadeSystem = new FacadeSystem();
        facadeSystem.methodA();
        facadeSystem.methodB();
        facadeSystem.methodC();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/facade/FacadePattern.kt
package kt.com.tencent.cain.desigin_pattern.facade

/**
 * @author : jiangyu
 * @date   : 2020/2/5
 * @desc   : 外观设计模式
 */
fun main() {
    val facadeSystem = FacadeSystem()
    facadeSystem.funcA()
    facadeSystem.funcB()
    facadeSystem.funcC()
}
